modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/security/FreeLDAPAuthenticationService.scala

```diff
@@ -0,0 +1,119 @@
+package edu.gemini.seqexec.web.server.security
+
+import com.unboundid.ldap.sdk._
+import edu.gemini.seqexec.web.common.UserDetails
+import edu.gemini.seqexec.web.server.security.AuthenticationService.AuthResult
+
+import scalaz.Scalaz._
+import scalaz._
+import scalaz.concurrent.Task
+import scalaz.effect.IO
+
+/**
+  * Definition of LDAP as a free monad algebra/interpreters
+  */
+object FreeLDAPAuthenticationService {
+  import UserDetails._
+  import LdapConnectionOps._
+
+  sealed trait LdapOp[A]
+  // Operations on ldap
+  object LdapOp {
+    // Operation to authenticate a user, It returns true if it works
+    case class AuthenticateOp(username: String, password: String) extends LdapOp[UID]
```

Comment:
Oh, initially it was a boolean, the comment is stale. I thought about just doing auth true/false but you need the uid for the next operations